src/ledger.js

```diff
@@ -682,18 +739,18 @@ export class LedgerExportExtendedPublicKey extends LedgerExportHDNode {
    * which need the fingerprint of the parent pubkey
    * @returns {string} fingerprint
    */
-   async getFingerprint() {
-    const pubkey = await this.getParentPublicKey()
-    return getFingerprintFromPublicKey(pubkey)
-   }
-  
+  async getFingerprint() {
+    const pubkey = await this.getParentPublicKey();
+    return getFingerprintFromPublicKey(pubkey);
+  }
+
   async getParentPublicKey() {
```

Comment:
Don't need this one either anymore.

Reply:
there's an await in the return

Reply:
So that should be covered by the async preceding the callback function. Since we're not awaiting anything in this scope, then it should be fine. 

It would be the same as if you wrote it like this:

```js
getParentPublicKey() {
  // just declaring the function here
  async function callbackFn(app) {
        const parentPath = getParentPath(this.bip32Path)	      
        return (await app.getWalletPublicKey(parentPath)).publicKey;
  }

  return this.withApp(callbackFn);
}
```

Reply:
This makes sense to me but let's chat tomorrow